.github/workflows/ci.yml

```diff
@@ -1,6 +1,7 @@
 name: CI
 
-on: [push]
+on:
+  pull_request:
```

Comment:
imo, `on: [push]` is still nice for testing branches before a PR is ready

Reply:
maybe we can add a `workflow_dispatch` also, so it can be triggered manually if needed?